src/redux/sagas/utils/proofs.ts

```diff
@@ -506,3 +524,58 @@ export function* getMultiPermissionProofs({
   // It does not need to be an array because if we get here, all the proofs are the same
   return proofs[0];
 }
+
+interface GetSinglePermissionProofsFromSourceDomainParams {
+  networkClient: ColonyNetworkClient;
+  colonyRoles: ColonyRoleFragment[];
+  colonyDomains: Domain[];
+  requiredDomainId: number;
+  requiredColonyRole: ColonyRole;
+  permissionAddress: string;
+  isMultiSig?: boolean;
+}
+
+// Returns the permission proofs from the domain where the user has been assigned the permissions
+// If the user has permissions directly in the requiredDomainId those will be returned first
+// However, if the user has inherited permissions in the requiredDomainId,
+// the permission proofs will be returned from the domain from which they are inherited
+export const getSinglePermissionProofsFromSourceDomain = async ({
```

Comment:
I know there's probably a reason for this, but at first glance, why would this be any different to how `getSinglePermissionProofsLocal` already works like?
That one tries to get permissions for the passed domainId, and if it doesn't find them it goes to the parent (root) domain and does it again for the parent domain

Reply:
This is actually part of the cherry-picked commit from #3145 so maybe better discussed over there, but the gist is that `getSinglePermissionProofsLocal` returns the permission proofs from the first domain you have permissions in, whereas this new function will return the permission proofs from the actual domain you have the permissions in. For example, you are funding an expenditure in Serenity, you have been assigned the necessary permissions in the root domain so you inherit them in Serenity. `getSinglePermissionProofsLocal` will return the permission proofs for Serenity, `getSinglePermissionProofsFromSourceDomain` will return the permission proofs for the root domain.

Reply:
ahhh gotcha, such an edge case, but in that case it makes sense :D and I would really prefer not to add another flag to that function, so you were spot on with decoupling it to a new function!
But it's still bugging me that it almost feels like `getSinglePermissionProofs` should function in the way you've described 
![image](https://github.com/user-attachments/assets/2da06d69-3eac-4657-80cb-0cc14a74bfe1)
but I guess the difference is with the `idx` return value?